From be3c7267f73a3db931a7f2c6ff64fada1dcac1e2 Mon Sep 17 00:00:00 2001
From: mortezadadgar <mortezadadgar97@gmail.com>
Date: Thu, 17 Apr 2025 14:20:23 +0330
Subject: [PATCH 5/7] allow reload xresources from signal

---
 x.c | 33 ++++++++++++++++++++++++++++++---
 1 file changed, 30 insertions(+), 3 deletions(-)

diff --git a/x.c b/x.c
index a54bb43..32ee9a6 100644
--- a/x.c
+++ b/x.c
@@ -2084,14 +2084,14 @@ resource_load(XrmDatabase db, char *name, enum resource_type rtype, void *dst)
 }
 
 void
-config_init(void)
+config_init(Display *dpy)
 {
 	char *resm;
 	XrmDatabase db;
 	ResourcePref *p;
 
 	XrmInitialize();
-	resm = XResourceManagerString(xw.dpy);
+	resm = XResourceManagerString(dpy);
 	if (!resm)
 		return;
 
@@ -2100,6 +2100,32 @@ config_init(void)
 		resource_load(db, p->name, p->type, p->dst);
 }
 
+void
+reload_config(int sig)
+{
+	/* Recreate a Display object to have up to date Xresources entries */
+	Display *dpy;
+	if (!(dpy = XOpenDisplay(NULL)))
+		die("Can't open display\n");
+
+	config_init(dpy);
+	xloadcols();
+
+	/* nearly like zoomabs() */
+	xunloadfonts();
+	xloadfonts(font, 0); /* font <- config_init() */
+	cresize(0, 0);
+	redraw();
+	xhints();
+
+	XCloseDisplay(dpy);
+
+	/* triggers re-render if we're visible */
+	ttywrite("\033[O", 3, 1);
+
+	signal(SIGUSR1, reload_config);
+}
+
 void
 usage(void)
 {
@@ -2177,7 +2203,8 @@ run:
 	if(!(xw.dpy = XOpenDisplay(NULL)))
 		die("Can't open display\n");
 
-	config_init();
+	config_init(xw.dpy);
+	signal(SIGUSR1, reload_config);
 	cols = MAX(cols, 1);
 	rows = MAX(rows, 1);
 	tnew(cols, rows);
-- 
2.49.0

